crates/electrum/src/electrum_ext.rs

```diff
@@ -165,20 +165,15 @@ pub trait ElectrumExt<A> {
     /// The scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
     /// transactions. `batch_size` specifies the max number of script pubkeys to request for in a
     /// single batch request.
-    fn scan<K: Ord + Clone>(
+    fn sync_with_keychain<K: Ord + Clone>(
         &self,
-        prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
```

Comment:
@LLFourn when you said:

> I would remove all the other arguments from scan_with_keychain other than the keychain_spks in both esplora and electrum. This emphasizes the special case that it should be used for.

Did you mean to remove *all* the other arguments, or just txids/outpoints? Personally I think it makes sense to leave `prev_tip`, `stop_gap`, `batch_size`, but I might be missing something

Reply:
Yes sorry @448-OG I meant what @danielabrozzoni. `scan_with_keychain` should only be for script pubkeys.